### PR TITLE
Fix data integrity issues causing missing contents in some post tooltips

### PR DESCRIPTION
**Pull request by oetherington**
There's three issues here that cause access checks to fail for some post revisions:
 - Some post contents revisions have a missing `collectionName`
 - Some post contents revisions have `draft` set to true, even though the relevant post is published
 - Some post contents revisions have a `version` < 1.0.0, even though the post is published

I missed these problems while testing as all of these permissions checks are bypassed when you login as an admin *facepalm*.

Note that these fields _are_ updated correctly for new posts, but the migration to denormalize the post contents field edited and created a lot of revisions, which is where these issues exist. The EA Forum has already run this migration so we will now need to run the new manual migration `undraftPublicPostRevisions`. LessWrong has not run the main migration yet, so this fix will be applied automatically for them without need to run the manual migration.

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1208054553428554) by [Unito](https://www.unito.io)

### CODE REVIEW

File: packages/lesswrong/server/manualMigrations/2024-08-14-undraftPublicRevisions.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import { getSqlClientOrThrow } from "../sql/sqlClient";
+import { registerMigration } from "./migrationUtils";
+import type { ITask } from "pg-promise";
+
+export const undraftPublicPostRevisions = async (db: SqlClient | ITask<{}>) => {
+  await db.none(`
+    UPDATE "Revisions" AS r
+    SET
+      "draft" = FALSE,
+      "version" = CASE WHEN LEFT("version", 1) = '0' THEN '1.0.0' ELSE "version" END
```
Comment (oetherington) [inline packages/lesswrong/server/manualMigrations/2024-08-14-undraftPublicRevisions.ts:10]: Undraft all revisions that are the current revision for a published post, and set the version to 1.0.0 if the current version is less than 1

File: packages/lesswrong/server/manualMigrations/2024-08-14-undraftPublicRevisions.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    FROM "Posts" AS p
+    WHERE
+      p."contents_latest" = r."_id" AND
+      p."draft" IS NOT TRUE AND
+      r."draft" IS TRUE
+  `);
+  await db.none(`
+    UPDATE "Revisions" AS r
+    SET "collectionName" = 'Posts'
```
Comment (oetherington) [inline packages/lesswrong/server/manualMigrations/2024-08-14-undraftPublicRevisions.ts:19]: Add any missing collection names

File: packages/lesswrong/server/migrations/meta/utils.ts [hunk 3]
```diff
@@ -307,18 +309,24 @@ export const normalizeEditableField = async <N extends CollectionNameString>(
...
-  `);
+  `, [collectionName]);
 
   // eslint-disable-next-line no-console
   console.log("Dropping denormalized field...");
   await dropRemovedField(db, collection, fieldName);
+
+  // eslint-disable-next-line no-console
+  console.log("Undrafting most recent revisions for published posts...");
+  await undraftPublicPostRevisions(db);
```
Comment (oetherington) [inline packages/lesswrong/server/migrations/meta/utils.ts:329]: EAF has already run this migration so we'll have to trigger it manually instead, but it'll run automatically for LW

Comment (Will-Howard) [reply packages/lesswrong/server/migrations/meta/utils.ts:329]: Is this necessary in addition to the change above? To me it reads like they are doing the same thing

**Review by Will-Howard — COMMENTED** [review]
ASTLAC

**Review by Will-Howard — APPROVED** [review]
Oops, didn't approve

File: packages/lesswrong/server/migrations/meta/utils.ts (hunk 3 above)
Comment (oetherington) [reply packages/lesswrong/server/migrations/meta/utils.ts:329]: Yeah - I'm just being overly cautious

**Pull request merged**